### DISCUSSION TITLE
Wallets Supporting eth_signTransaction Listed on WalletConnect

### DISCUSSION THREAD

**Discussion by judeProground** [WalletConnect/walletconnect-monorepo#4239] [Developer Support] [answered]
Hello.
Is there a way to know all types of wallets listed on WalletConnect that support "eth_signTransaction" rpc method? 
I know MetaMask does not support "eth_signTransaction" for security reasons, but I'm curious about whether other wallets support it.

Thanks!

**Comment by glitch-txs** [comment]
you should check this in the session response receive by the wallet after connecting, in the case of ethereum-provider you can find this in `provider.session`

**Reply by judeProground** [reply]
Thank you for your response, glitch-txs!
Does it mean that I should check the session response after connecting to each wallet?

**Reply by glitch-txs** [reply, marked as answer]
you could, most wallets don't support this btw, if you're looking for a normal signature you can use personal_sign or signTypedData_v4
